src/charm.py

```diff
@@ -123,6 +129,7 @@ def _process_wheel_file(self, event):  # noqa: C901
 
         try:
             resource_path = self.model.resources.fetch("workflows-file")
+            filename = str(resource_path).rsplit("/", maxsplit=1)
```

Comment:
There might be a problem here. `rsplit` returns a list. Did I miss something?
A better way of doing this might be to use `pathlib`

```python
from pathlib import Path
filename = Path(resource_path).name
```

Reply:
Updated, thanks!